Assets/_GameJamCat/Scripts/UI/EndGameMenu.cs

```diff
@@ -0,0 +1,78 @@
+using UnityEngine;
+using Sirenix.OdinInspector;
+using UnityEngine.UI;
+
+namespace GameJamCat
+{
+    public class EndGameMenu : MonoBehaviour
+    {
+        [SerializeField] private GameObject _loseScreen = null;
+        [SerializeField] private GameObject _winScreen = null;
+        [SerializeField] private Button _loseReplayButton = null;
+        [SerializeField] private Button _winReplayButton = null;
+
+        private bool _foundCat = false; // TempCondition until manager has information
+        
+        [Button]
+        private void WinGame()
+        {
+            _foundCat = true;
+            StateManager.Instance.SetState(State.EndGame);
+        }
+        
+        [Button]
+        private void LoseGame()
+        {
+            _foundCat = false;
+            StateManager.Instance.SetState(State.EndGame);
+        }
+
+        private void Awake()
+        {
+            if (_loseScreen != null)
+            {
+                _loseScreen.SetActive(false);
+            }
+
+            if (_winScreen != null)
+            {
+                _winScreen.SetActive(false);
+            }
+
+            StateManager.Instance.OnStateChanged += DisplayEndPanel;
+            _loseReplayButton.onClick.AddListener(() => ResetGame());
+            _winReplayButton.onClick.AddListener(() => ResetGame());
+        }
+
+        private void OnDestroy()
+        {
+            StateManager.Instance.OnStateChanged -= DisplayEndPanel;
+        }
+
+        private void DisplayEndPanel(State state)
```

Comment:
This should be a public method. 

Ideally this is called from UIManager, since it's the WorldManager that knows if it's the end of the game,  sent to UIManager, and then UIManager will tell this EndGameMenu to "DisplayEndPanel()"

Maybe you can pass in the _foundCat in to here. So it's DisplayEndPanel(Bool foundCat)

so that this view knows no logic.  It just knows to display it.

Reply:
Fair enough, sounds good!

So just to be clear, then there is no reason for this script to subscribe to the Statemanager right? Since the UIManager will handle that now!

Reply:
exactly